src/credentials_rotators/npm-token-rotation/README.md

```diff
@@ -0,0 +1,112 @@
+| npmLoginUsernameSecret                                           | NPM Login name `arn` and `secretKey`                                                                 |
+| npmLoginPasswordSecret                                           | NPM Login password `arn` and `secretKey`                                                             |
+| npmOtpSeedSecret                                                 | TOTP Seed `arn` and `secretKey`                                                                      |
+| npmAccessTokenSecrets.secrets                                    | List of tokens that needs rotation                                                                   |
+| npmAccessTokenSecrets.secrets[].arn                              | ARN of NPM token that needs rotation                                                                 |
+| npmAccessTokenSecrets.secrets[].secretKey                        | Secret Key                                                                                           |
+| npmAccessTokenSecrets.secrets[].publishConfig.type               | Location where environment variable gets stored. This could either be Context or Environment         |
+| npmAccessTokenSecrets.secrets[].publishConfig.contextName        | Name of CircleCI context when type is Context                                                        |
+| npmAccessTokenSecrets.secrets[].publishConfig.projectName        | Name of CircleCI project where the environment variable is when type stored when type is Environment |
+| npmAccessTokenSecrets.secrets[].publishConfig.slug               | CircleCI Slug. For GitHub it starts with gh and username. Examples are `gh/aws-amplify` or `gh/yuth` |
+| npmAccessTokenSecrets.secrets[].publishConfig.variableName       | Environment variable name                                                                            |
+| npmAccessTokenSecrets.secrets[].publishConfig.circleCIToken      | CircleCI Token secret `arn` and `secretKey`                                                          |
+| npmAccessTokenSecrets.secrets[].publishConfig.slackWebHookConfig | [Optional] `Arn` and `secretKey` for slackWebhookUrl used for notification                           |
+
+## High Level Architecture
+![](./docs/arch.svg)
+
+The CDK stack adds a secret rotation lambda function which would trigger automatic rotation of Configured NPM Token. When the automatic rotation [lambda function](src/lambda/create-new-token/index.ts) is triggered which requests creation of new NPM token using the credentials provided in the `secret_config.json` and triggers Step Function to which [pushes the credentials to CircleCI](src/lambda/step-01-publish-token/index.ts) and after 15 minutes of wait it [deletes the old tokens](src/lambda/step-02-delete-old-token/index.ts).
```

Comment:
probably a good idea to specify the wait time in the config to suit different workflows?

Reply:
We re-use the same Step Function, so making it per secret might not be possible as step function wait has to be part of the function definition. But we can provide a global level config